WordPress/Classes/Utility/Logging/WPCrashLoggingProvider.swift

```diff
@@ -23,4 +34,92 @@ class WPCrashLoggingProvider: CrashLoggingDataProvider {
 
         return TracksUser(userID: account.userID.stringValue, email: account.email, username: account.username)
     }
+
+    var loggingUploadDelegate: EventLoggingDelegate {
+        return self
```

Comment:
I'm not particularly tied to this implementation 🤷 

It makes the AppDelegate mildly more complex, but that isn't a huge deal.

We already had a way to inspect `userHasOptedOut` statically, so I'm just using that instead of passing it around or using closures: f71277c22de4170251ae9c701b495b81945453ab.

Reply:
Thank you! I assume this is just moving pieces around, so I am not re-reviewing this bit.